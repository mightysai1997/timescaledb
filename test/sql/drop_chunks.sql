\o /dev/null
\ir include/create_single_db.sql
\o

CREATE TABLE PUBLIC.drop_chunk_test1(time bigint, temp float8, device_id text);
CREATE TABLE PUBLIC.drop_chunk_test2(time bigint, temp float8, device_id text);
CREATE INDEX ON drop_chunk_test1(time DESC);
SELECT create_hypertable('public.drop_chunk_test1', 'time', chunk_time_interval => 1, create_default_indexes=>false);
SELECT create_hypertable('public.drop_chunk_test2', 'time', chunk_time_interval => 1, create_default_indexes=>false);

SELECT c.id AS chunk_id, c.partition_id, pe.hypertable_id, c.schema_name AS chunk_schema, c.table_name AS chunk_table, c.start_time, c.end_time
FROM _timescaledb_catalog.chunk c
INNER JOIN _timescaledb_catalog.partition p ON (p.id = c.partition_id)
INNER JOIN _timescaledb_catalog.partition_epoch pe ON (pe.id = p.epoch_id)
INNER JOIN _timescaledb_catalog.hypertable h ON (pe.hypertable_id = h.id)
WHERE h.schema_name = 'public' AND (h.table_name = 'drop_chunk_test1' OR h.table_name = 'drop_chunk_test2');

\dt "_timescaledb_internal".*

SELECT  _timescaledb_internal.get_partition_for_key('dev1', 32768);
SELECT  _timescaledb_internal.get_partition_for_key('dev7', 32768);

INSERT INTO PUBLIC.drop_chunk_test1 VALUES(1, 1.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test1 VALUES(2, 2.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test1 VALUES(3, 3.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test1 VALUES(4, 4.0, 'dev7');
INSERT INTO PUBLIC.drop_chunk_test1 VALUES(5, 5.0, 'dev7');
INSERT INTO PUBLIC.drop_chunk_test1 VALUES(6, 6.0, 'dev7');

INSERT INTO PUBLIC.drop_chunk_test2 VALUES(1, 1.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test2 VALUES(2, 2.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test2 VALUES(3, 3.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test2 VALUES(4, 4.0, 'dev7');
INSERT INTO PUBLIC.drop_chunk_test2 VALUES(5, 5.0, 'dev7');
INSERT INTO PUBLIC.drop_chunk_test2 VALUES(6, 6.0, 'dev7');

SELECT c.id AS chunk_id, c.partition_id, pe.hypertable_id, c.schema_name AS chunk_schema, c.table_name AS chunk_table, c.start_time, c.end_time
FROM _timescaledb_catalog.chunk c
INNER JOIN _timescaledb_catalog.partition p ON (p.id = c.partition_id)
INNER JOIN _timescaledb_catalog.partition_epoch pe ON (pe.id = p.epoch_id)
INNER JOIN _timescaledb_catalog.hypertable h ON (pe.hypertable_id = h.id)
WHERE h.schema_name = 'public' AND (h.table_name = 'drop_chunk_test1' OR h.table_name = 'drop_chunk_test2');

\dt "_timescaledb_internal".*

SELECT _timescaledb_internal.drop_chunks_older_than(2);

SELECT c.id AS chunk_id, c.partition_id, pe.hypertable_id, c.schema_name AS chunk_schema, c.table_name AS chunk_table, c.start_time, c.end_time
FROM _timescaledb_catalog.chunk c
INNER JOIN _timescaledb_catalog.partition p ON (p.id = c.partition_id)
INNER JOIN _timescaledb_catalog.partition_epoch pe ON (pe.id = p.epoch_id)
INNER JOIN _timescaledb_catalog.hypertable h ON (pe.hypertable_id = h.id)
WHERE h.schema_name = 'public' AND (h.table_name = 'drop_chunk_test1' OR h.table_name = 'drop_chunk_test2');

\dt "_timescaledb_internal".*

SELECT _timescaledb_internal.drop_chunks_older_than(3, 'drop_chunk_test1');

SELECT c.id AS chunk_id, c.partition_id, pe.hypertable_id, c.schema_name AS chunk_schema, c.table_name AS chunk_table, c.start_time, c.end_time
FROM _timescaledb_catalog.chunk c
INNER JOIN _timescaledb_catalog.partition p ON (p.id = c.partition_id)
INNER JOIN _timescaledb_catalog.partition_epoch pe ON (pe.id = p.epoch_id)
INNER JOIN _timescaledb_catalog.hypertable h ON (pe.hypertable_id = h.id)
WHERE h.schema_name = 'public' AND (h.table_name = 'drop_chunk_test1' OR h.table_name = 'drop_chunk_test2');

\dt "_timescaledb_internal".*
