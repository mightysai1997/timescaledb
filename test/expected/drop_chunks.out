CREATE TABLE PUBLIC.drop_chunk_test1(time bigint, temp float8, device_id text);
CREATE TABLE PUBLIC.drop_chunk_test2(time bigint, temp float8, device_id text);
CREATE TABLE PUBLIC.drop_chunk_test3(time bigint, temp float8, device_id text);
CREATE INDEX ON drop_chunk_test1(time DESC);
SELECT create_hypertable('public.drop_chunk_test1', 'time', chunk_time_interval => 1, create_default_indexes=>false);
NOTICE:  adding NOT NULL constraint to column "time"
 create_hypertable 
-------------------
 
(1 row)

SELECT create_hypertable('public.drop_chunk_test2', 'time', chunk_time_interval => 1, create_default_indexes=>false);
NOTICE:  adding NOT NULL constraint to column "time"
 create_hypertable 
-------------------
 
(1 row)

SELECT create_hypertable('public.drop_chunk_test3', 'time', chunk_time_interval => 1, create_default_indexes=>false);
NOTICE:  adding NOT NULL constraint to column "time"
 create_hypertable 
-------------------
 
(1 row)

SELECT c.id AS chunk_id, c.hypertable_id, c.schema_name AS chunk_schema, c.table_name AS chunk_table, ds.range_start, ds.range_end
FROM _timescaledb_catalog.chunk c
INNER JOIN _timescaledb_catalog.hypertable h ON (c.hypertable_id = h.id)
INNER JOIN  _timescaledb_internal.dimension_get_time(h.id) time_dimension ON(true)
INNER JOIN  _timescaledb_catalog.dimension_slice ds ON (ds.dimension_id = time_dimension.id)
INNER JOIN  _timescaledb_catalog.chunk_constraint cc ON (cc.dimension_slice_id = ds.id AND cc.chunk_id = c.id)
WHERE h.schema_name = 'public' AND (h.table_name = 'drop_chunk_test1' OR h.table_name = 'drop_chunk_test2');
 chunk_id | hypertable_id | chunk_schema | chunk_table | range_start | range_end 
----------+---------------+--------------+-------------+-------------+-----------
(0 rows)

\dt "_timescaledb_internal".*
      List of relations
 Schema | Name | Type | Owner 
--------+------+------+-------
(0 rows)

SELECT  _timescaledb_internal.get_partition_for_key('dev1'::text);
 get_partition_for_key 
-----------------------
            1129986420
(1 row)

SELECT  _timescaledb_internal.get_partition_for_key('dev7'::varchar(5));
 get_partition_for_key 
-----------------------
             449729092
(1 row)

INSERT INTO PUBLIC.drop_chunk_test1 VALUES(1, 1.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test1 VALUES(2, 2.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test1 VALUES(3, 3.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test1 VALUES(4, 4.0, 'dev7');
INSERT INTO PUBLIC.drop_chunk_test1 VALUES(5, 5.0, 'dev7');
INSERT INTO PUBLIC.drop_chunk_test1 VALUES(6, 6.0, 'dev7');
INSERT INTO PUBLIC.drop_chunk_test2 VALUES(1, 1.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test2 VALUES(2, 2.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test2 VALUES(3, 3.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test2 VALUES(4, 4.0, 'dev7');
INSERT INTO PUBLIC.drop_chunk_test2 VALUES(5, 5.0, 'dev7');
INSERT INTO PUBLIC.drop_chunk_test2 VALUES(6, 6.0, 'dev7');
INSERT INTO PUBLIC.drop_chunk_test3 VALUES(1, 1.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test3 VALUES(2, 2.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test3 VALUES(3, 3.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test3 VALUES(4, 4.0, 'dev7');
INSERT INTO PUBLIC.drop_chunk_test3 VALUES(5, 5.0, 'dev7');
INSERT INTO PUBLIC.drop_chunk_test3 VALUES(6, 6.0, 'dev7');
SELECT c.id AS chunk_id, c.hypertable_id, c.schema_name AS chunk_schema, c.table_name AS chunk_table, ds.range_start, ds.range_end
FROM _timescaledb_catalog.chunk c
INNER JOIN _timescaledb_catalog.hypertable h ON (c.hypertable_id = h.id)
INNER JOIN  _timescaledb_internal.dimension_get_time(h.id) time_dimension ON(true)
INNER JOIN  _timescaledb_catalog.dimension_slice ds ON (ds.dimension_id = time_dimension.id)
INNER JOIN  _timescaledb_catalog.chunk_constraint cc ON (cc.dimension_slice_id = ds.id AND cc.chunk_id = c.id)
WHERE h.schema_name = 'public' AND (h.table_name = 'drop_chunk_test1' OR h.table_name = 'drop_chunk_test2');
 chunk_id | hypertable_id |     chunk_schema      |    chunk_table    | range_start | range_end 
----------+---------------+-----------------------+-------------------+-------------+-----------
        1 |             1 | _timescaledb_internal | _hyper_1_1_chunk  |           1 |         2
        2 |             1 | _timescaledb_internal | _hyper_1_2_chunk  |           2 |         3
        3 |             1 | _timescaledb_internal | _hyper_1_3_chunk  |           3 |         4
        4 |             1 | _timescaledb_internal | _hyper_1_4_chunk  |           4 |         5
        5 |             1 | _timescaledb_internal | _hyper_1_5_chunk  |           5 |         6
        6 |             1 | _timescaledb_internal | _hyper_1_6_chunk  |           6 |         7
        7 |             2 | _timescaledb_internal | _hyper_2_7_chunk  |           1 |         2
        8 |             2 | _timescaledb_internal | _hyper_2_8_chunk  |           2 |         3
        9 |             2 | _timescaledb_internal | _hyper_2_9_chunk  |           3 |         4
       10 |             2 | _timescaledb_internal | _hyper_2_10_chunk |           4 |         5
       11 |             2 | _timescaledb_internal | _hyper_2_11_chunk |           5 |         6
       12 |             2 | _timescaledb_internal | _hyper_2_12_chunk |           6 |         7
(12 rows)

\dt "_timescaledb_internal".*
                           List of relations
        Schema         |       Name        | Type  |       Owner       
-----------------------+-------------------+-------+-------------------
 _timescaledb_internal | _hyper_1_1_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_1_2_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_1_3_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_1_4_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_1_5_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_1_6_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_2_10_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_2_11_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_2_12_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_2_7_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_2_8_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_2_9_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_3_13_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_3_14_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_3_15_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_3_16_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_3_17_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_3_18_chunk | table | default_perm_user
(18 rows)

CREATE VIEW dependent_view AS SELECT * FROM _timescaledb_internal._hyper_1_1_chunk;
\set ON_ERROR_STOP 0
SELECT drop_chunks(2);
ERROR:  cannot drop table _timescaledb_internal._hyper_1_1_chunk because other objects depend on it
\set ON_ERROR_STOP 1
SELECT drop_chunks(2, CASCADE=>true);
NOTICE:  drop cascades to view dependent_view
 drop_chunks 
-------------
 
(1 row)

SELECT c.id AS chunk_id, c.hypertable_id, c.schema_name AS chunk_schema, c.table_name AS chunk_table, ds.range_start, ds.range_end
FROM _timescaledb_catalog.chunk c
INNER JOIN _timescaledb_catalog.hypertable h ON (c.hypertable_id = h.id)
INNER JOIN  _timescaledb_internal.dimension_get_time(h.id) time_dimension ON(true)
INNER JOIN  _timescaledb_catalog.dimension_slice ds ON (ds.dimension_id = time_dimension.id)
INNER JOIN  _timescaledb_catalog.chunk_constraint cc ON (cc.dimension_slice_id = ds.id AND cc.chunk_id = c.id)
WHERE h.schema_name = 'public' AND (h.table_name = 'drop_chunk_test1' OR h.table_name = 'drop_chunk_test2');
 chunk_id | hypertable_id |     chunk_schema      |    chunk_table    | range_start | range_end 
----------+---------------+-----------------------+-------------------+-------------+-----------
        2 |             1 | _timescaledb_internal | _hyper_1_2_chunk  |           2 |         3
        3 |             1 | _timescaledb_internal | _hyper_1_3_chunk  |           3 |         4
        4 |             1 | _timescaledb_internal | _hyper_1_4_chunk  |           4 |         5
        5 |             1 | _timescaledb_internal | _hyper_1_5_chunk  |           5 |         6
        6 |             1 | _timescaledb_internal | _hyper_1_6_chunk  |           6 |         7
        8 |             2 | _timescaledb_internal | _hyper_2_8_chunk  |           2 |         3
        9 |             2 | _timescaledb_internal | _hyper_2_9_chunk  |           3 |         4
       10 |             2 | _timescaledb_internal | _hyper_2_10_chunk |           4 |         5
       11 |             2 | _timescaledb_internal | _hyper_2_11_chunk |           5 |         6
       12 |             2 | _timescaledb_internal | _hyper_2_12_chunk |           6 |         7
(10 rows)

\dt "_timescaledb_internal".*
                           List of relations
        Schema         |       Name        | Type  |       Owner       
-----------------------+-------------------+-------+-------------------
 _timescaledb_internal | _hyper_1_2_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_1_3_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_1_4_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_1_5_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_1_6_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_2_10_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_2_11_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_2_12_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_2_8_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_2_9_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_3_14_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_3_15_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_3_16_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_3_17_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_3_18_chunk | table | default_perm_user
(15 rows)

SELECT drop_chunks(3, 'drop_chunk_test1');
 drop_chunks 
-------------
 
(1 row)

SELECT c.id AS chunk_id, c.hypertable_id, c.schema_name AS chunk_schema, c.table_name AS chunk_table, ds.range_start, ds.range_end
FROM _timescaledb_catalog.chunk c
INNER JOIN _timescaledb_catalog.hypertable h ON (c.hypertable_id = h.id)
INNER JOIN  _timescaledb_internal.dimension_get_time(h.id) time_dimension ON(true)
INNER JOIN  _timescaledb_catalog.dimension_slice ds ON (ds.dimension_id = time_dimension.id)
INNER JOIN  _timescaledb_catalog.chunk_constraint cc ON (cc.dimension_slice_id = ds.id AND cc.chunk_id = c.id)
WHERE h.schema_name = 'public' AND (h.table_name = 'drop_chunk_test1' OR h.table_name = 'drop_chunk_test2');
 chunk_id | hypertable_id |     chunk_schema      |    chunk_table    | range_start | range_end 
----------+---------------+-----------------------+-------------------+-------------+-----------
        3 |             1 | _timescaledb_internal | _hyper_1_3_chunk  |           3 |         4
        4 |             1 | _timescaledb_internal | _hyper_1_4_chunk  |           4 |         5
        5 |             1 | _timescaledb_internal | _hyper_1_5_chunk  |           5 |         6
        6 |             1 | _timescaledb_internal | _hyper_1_6_chunk  |           6 |         7
        8 |             2 | _timescaledb_internal | _hyper_2_8_chunk  |           2 |         3
        9 |             2 | _timescaledb_internal | _hyper_2_9_chunk  |           3 |         4
       10 |             2 | _timescaledb_internal | _hyper_2_10_chunk |           4 |         5
       11 |             2 | _timescaledb_internal | _hyper_2_11_chunk |           5 |         6
       12 |             2 | _timescaledb_internal | _hyper_2_12_chunk |           6 |         7
(9 rows)

\dt "_timescaledb_internal".*
                           List of relations
        Schema         |       Name        | Type  |       Owner       
-----------------------+-------------------+-------+-------------------
 _timescaledb_internal | _hyper_1_3_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_1_4_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_1_5_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_1_6_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_2_10_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_2_11_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_2_12_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_2_8_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_2_9_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_3_14_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_3_15_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_3_16_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_3_17_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_3_18_chunk | table | default_perm_user
(14 rows)

-- 2,147,483,647 is the largest int so this tests that BIGINTs work
SELECT drop_chunks(2147483648, 'drop_chunk_test3');
 drop_chunks 
-------------
 
(1 row)

SELECT c.id AS chunk_id, c.hypertable_id, c.schema_name AS chunk_schema, c.table_name AS chunk_table, ds.range_start, ds.range_end
FROM _timescaledb_catalog.chunk c
INNER JOIN _timescaledb_catalog.hypertable h ON (c.hypertable_id = h.id)
INNER JOIN  _timescaledb_internal.dimension_get_time(h.id) time_dimension ON(true)
INNER JOIN  _timescaledb_catalog.dimension_slice ds ON (ds.dimension_id = time_dimension.id)
INNER JOIN  _timescaledb_catalog.chunk_constraint cc ON (cc.dimension_slice_id = ds.id AND cc.chunk_id = c.id)
WHERE h.schema_name = 'public' AND (h.table_name = 'drop_chunk_test1' OR h.table_name = 'drop_chunk_test2' OR h.table_name = 'drop_chunk_test3');
 chunk_id | hypertable_id |     chunk_schema      |    chunk_table    | range_start | range_end 
----------+---------------+-----------------------+-------------------+-------------+-----------
        3 |             1 | _timescaledb_internal | _hyper_1_3_chunk  |           3 |         4
        4 |             1 | _timescaledb_internal | _hyper_1_4_chunk  |           4 |         5
        5 |             1 | _timescaledb_internal | _hyper_1_5_chunk  |           5 |         6
        6 |             1 | _timescaledb_internal | _hyper_1_6_chunk  |           6 |         7
        8 |             2 | _timescaledb_internal | _hyper_2_8_chunk  |           2 |         3
        9 |             2 | _timescaledb_internal | _hyper_2_9_chunk  |           3 |         4
       10 |             2 | _timescaledb_internal | _hyper_2_10_chunk |           4 |         5
       11 |             2 | _timescaledb_internal | _hyper_2_11_chunk |           5 |         6
       12 |             2 | _timescaledb_internal | _hyper_2_12_chunk |           6 |         7
(9 rows)

\dt "_timescaledb_internal".*
                           List of relations
        Schema         |       Name        | Type  |       Owner       
-----------------------+-------------------+-------+-------------------
 _timescaledb_internal | _hyper_1_3_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_1_4_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_1_5_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_1_6_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_2_10_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_2_11_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_2_12_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_2_8_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_2_9_chunk  | table | default_perm_user
(9 rows)

-- should error because no hypertable
\set ON_ERROR_STOP 0
SELECT drop_chunks(5, 'drop_chunk_test4');
ERROR:  No hypertables found
\set ON_ERROR_STOP 1
DROP TABLE _timescaledb_internal._hyper_1_6_chunk;
SELECT c.id AS chunk_id, c.hypertable_id, c.schema_name AS chunk_schema, c.table_name AS chunk_table, ds.range_start, ds.range_end
FROM _timescaledb_catalog.chunk c
INNER JOIN _timescaledb_catalog.hypertable h ON (c.hypertable_id = h.id)
INNER JOIN  _timescaledb_internal.dimension_get_time(h.id) time_dimension ON(true)
INNER JOIN  _timescaledb_catalog.dimension_slice ds ON (ds.dimension_id = time_dimension.id)
INNER JOIN  _timescaledb_catalog.chunk_constraint cc ON (cc.dimension_slice_id = ds.id AND cc.chunk_id = c.id)
WHERE h.schema_name = 'public' AND (h.table_name = 'drop_chunk_test1' OR h.table_name = 'drop_chunk_test2');
 chunk_id | hypertable_id |     chunk_schema      |    chunk_table    | range_start | range_end 
----------+---------------+-----------------------+-------------------+-------------+-----------
        3 |             1 | _timescaledb_internal | _hyper_1_3_chunk  |           3 |         4
        4 |             1 | _timescaledb_internal | _hyper_1_4_chunk  |           4 |         5
        5 |             1 | _timescaledb_internal | _hyper_1_5_chunk  |           5 |         6
        8 |             2 | _timescaledb_internal | _hyper_2_8_chunk  |           2 |         3
        9 |             2 | _timescaledb_internal | _hyper_2_9_chunk  |           3 |         4
       10 |             2 | _timescaledb_internal | _hyper_2_10_chunk |           4 |         5
       11 |             2 | _timescaledb_internal | _hyper_2_11_chunk |           5 |         6
       12 |             2 | _timescaledb_internal | _hyper_2_12_chunk |           6 |         7
(8 rows)

\dt "_timescaledb_internal".*
                           List of relations
        Schema         |       Name        | Type  |       Owner       
-----------------------+-------------------+-------+-------------------
 _timescaledb_internal | _hyper_1_3_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_1_4_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_1_5_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_2_10_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_2_11_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_2_12_chunk | table | default_perm_user
 _timescaledb_internal | _hyper_2_8_chunk  | table | default_perm_user
 _timescaledb_internal | _hyper_2_9_chunk  | table | default_perm_user
(8 rows)

CREATE TABLE PUBLIC.drop_chunk_test_ts(time timestamp, temp float8, device_id text);
SELECT create_hypertable('public.drop_chunk_test_ts', 'time', chunk_time_interval => interval '1 minute', create_default_indexes=>false);
NOTICE:  adding NOT NULL constraint to column "time"
 create_hypertable 
-------------------
 
(1 row)

CREATE TABLE PUBLIC.drop_chunk_test_tstz(time timestamptz, temp float8, device_id text);
SELECT create_hypertable('public.drop_chunk_test_tstz', 'time', chunk_time_interval => interval '1 minute', create_default_indexes=>false);
NOTICE:  adding NOT NULL constraint to column "time"
 create_hypertable 
-------------------
 
(1 row)

SET timezone = '+1';
INSERT INTO PUBLIC.drop_chunk_test_ts VALUES(now()-INTERVAL '5 minutes', 1.0, 'dev1');
INSERT INTO PUBLIC.drop_chunk_test_tstz VALUES(now()-INTERVAL '5 minutes', 1.0, 'dev1');
SELECT * FROM test.show_subtables('drop_chunk_test_ts');
                  Child                  | Tablespace 
-----------------------------------------+------------
 _timescaledb_internal._hyper_4_19_chunk | 
(1 row)

SELECT * FROM test.show_subtables('drop_chunk_test_tstz');
                  Child                  | Tablespace 
-----------------------------------------+------------
 _timescaledb_internal._hyper_5_20_chunk | 
(1 row)

BEGIN;
    SELECT drop_chunks(interval '1 minute', 'drop_chunk_test_ts');
 drop_chunks 
-------------
 
(1 row)

    SELECT * FROM test.show_subtables('drop_chunk_test_ts');
 Child | Tablespace 
-------+------------
(0 rows)

    SELECT drop_chunks(interval '1 minute', 'drop_chunk_test_tstz');
 drop_chunks 
-------------
 
(1 row)

    SELECT * FROM test.show_subtables('drop_chunk_test_tstz');
 Child | Tablespace 
-------+------------
(0 rows)

ROLLBACK;
BEGIN;
    SELECT drop_chunks(now()::timestamp-interval '1 minute', 'drop_chunk_test_ts');
 drop_chunks 
-------------
 
(1 row)

    SELECT * FROM test.show_subtables('drop_chunk_test_ts');
 Child | Tablespace 
-------+------------
(0 rows)

    SELECT drop_chunks(now()-interval '1 minute', 'drop_chunk_test_tstz');
 drop_chunks 
-------------
 
(1 row)

    SELECT * FROM test.show_subtables('drop_chunk_test_tstz');
 Child | Tablespace 
-------+------------
(0 rows)

ROLLBACK;
\set ON_ERROR_STOP 0
SELECT drop_chunks(interval '1 minute');
ERROR:  Cannot use drop_chunks on multiple tables with different time types
SELECT drop_chunks(interval '1 minute', 'drop_chunk_test3');
ERROR:  Can only use drop_chunks with an INTERVAL for TIMESTAMP, TIMESTAMPTZ, and DATE types
SELECT drop_chunks(now()-interval '1 minute', 'drop_chunk_test_ts');
ERROR:  Cannot call drop_chunks with a timestamp with time zone on hypertables with a time type of: timestamp without time zone
SELECT drop_chunks(now()::timestamp-interval '1 minute', 'drop_chunk_test_tstz');
ERROR:  Cannot call drop_chunks with a timestamp without time zone on hypertables with a time type of: timestamp with time zone
\set ON_ERROR_STOP 1
CREATE TABLE PUBLIC.drop_chunk_test_date(time date, temp float8, device_id text);
SELECT create_hypertable('public.drop_chunk_test_date', 'time', chunk_time_interval => interval '1 day', create_default_indexes=>false);
NOTICE:  adding NOT NULL constraint to column "time"
 create_hypertable 
-------------------
 
(1 row)

SET timezone = '+100';
INSERT INTO PUBLIC.drop_chunk_test_date VALUES(now()-INTERVAL '2 day', 1.0, 'dev1');
BEGIN;
    SELECT drop_chunks(interval '1 day', 'drop_chunk_test_date');
 drop_chunks 
-------------
 
(1 row)

    SELECT * FROM test.show_subtables('drop_chunk_test_date');
 Child | Tablespace 
-------+------------
(0 rows)

ROLLBACK;
BEGIN;
    SELECT drop_chunks((now()-interval '1 day')::date, 'drop_chunk_test_date');
 drop_chunks 
-------------
 
(1 row)

    SELECT * FROM test.show_subtables('drop_chunk_test_date');
 Child | Tablespace 
-------+------------
(0 rows)

ROLLBACK;
